src/database/migrations/0055-create-audit-log.sql

```diff
@@ -0,0 +1,439 @@
+	AND TG_LEVEL = 'ROW'
+) THEN audit_row.row_data = row_to_json(OLD)::JSONB - excluded_cols;
+--Computing differences
+SELECT jsonb_object_agg(tmp_new_row.key, tmp_new_row.value) AS new_data
+INTO audit_row.changed_fields
+FROM jsonb_each_text(row_to_json(NEW)::JSONB) AS tmp_new_row
+	JOIN jsonb_each_text(audit_row.row_data) AS tmp_old_row ON (
+		tmp_new_row.key = tmp_old_row.key
+		AND tmp_new_row.value IS DISTINCT
+		FROM tmp_old_row.value
+	);
+IF audit_row.changed_fields = '{}'::JSONB THEN
+-- All changed fields are ignored. Skip this update.
+RETURN NULL;
+END IF;
+ELSIF (
+	TG_OP = 'DELETE'
+	AND TG_LEVEL = 'ROW'
+) THEN audit_row.row_data = row_to_json(OLD)::JSONB - excluded_cols;
+ELSIF (
+	TG_OP = 'INSERT'
+	AND TG_LEVEL = 'ROW'
+) THEN audit_row.row_data = row_to_json(NEW)::JSONB - excluded_cols;
+ELSIF (
+	TG_LEVEL = 'STATEMENT'
+	AND TG_OP IN ('INSERT', 'UPDATE', 'DELETE', 'TRUNCATE')
+) THEN audit_row.statement_only = 't';
+ELSE RAISE EXCEPTION
+'[audit_if_modified_func] - Trigger func added as trigger for unhandled case: %, %',
+TG_OP,
+TG_LEVEL;
+RETURN NULL;
+END IF;
+INSERT INTO audit_logs
+VALUES (audit_row.*);
+RETURN NULL;
+END;
+$body$ LANGUAGE plpgsql;
+COMMENT ON FUNCTION audit_if_modified_func() IS $body$ Track changes to a table
+ at the statement and / or row level.Optional parameters to trigger in CREATE
+ TRIGGER call: param 0: boolean, whether to log the query text.Default
+ 't'.param 1: text [], columns to ignore in updates.Default [].Updates to
+ ignored cols are omitted from changed_fields.Updates with only ignored cols
+ changed are not inserted into the audit log.Almost all the processing work is
+ still done for updates that ignored.If you need to save the load, you need to
+ use WHEN clause on the trigger instead. No warning	or error is issued if
+ ignored_cols contains columns that do not exist in the target table.This lets
+ you specify a standard set of ignored columns. There is no parameter to
+ disable logging of values. Add this trigger as a 'FOR EACH STATEMENT' rather
+ than 'FOR EACH ROW' trigger if you do not want to log row values.Note that the
+ user name logged is the login role for the session.The audit trigger cannot
+ obtain the active role because it is reset by the SECURITY DEFINER invocation
+ of the audit trigger its self.$body$;
+
+CREATE OR REPLACE FUNCTION audit_table(
+	target_table regclass,
+	audit_rows boolean,
+	audit_query_text boolean,
+	audit_inserts boolean,
+	ignored_cols text []
+) RETURNS void AS $body$
+DECLARE stm_targets text = 'INSERT OR UPDATE OR DELETE OR TRUNCATE';
+_q_txt text;
+_ignored_cols_snip text = '';
+BEGIN PERFORM deaudit_table(target_table);
+-- Log the begin audit event into the audit table.
+INSERT INTO audit_logs (
+	event_id,
+	schema_name,
+	table_name,
+	relid,
+	session_user_name,
+	action_tstamp_tx,
+	action_tstamp_stm,
+	action_tstamp_clk,
+	transaction_id,
+	application_name,
+	client_addr,
+	client_port,
+	client_query,
+	action,
+	statement_only)
```

Comment:
Given that some calls can have filters on them, it would be useful to include those args in the `row_data` so that we can see whether to expect certain changes or not.